### DISCUSSION TITLE
Vercel Build failing with latest PNPM Version

### DISCUSSION THREAD

**Discussion by z4nr34l** [vercel/community#2641] [Bug Report] [answered]
### Page to Investigate

https://vercel.com/templates/next.js/nextjs-boilerplate

### Steps to Reproduce

1. Deploy Next.js App with pnpm `8.5.1` - which would just go through
2. Upgrade pnpm to `8.6.0`
3. Reinstall dependencies - there is change, that upgrades lockfile from `6.0` to `6.1`
4. Next deployment will fail with error `ERR_PNPM_LOCKFILE_MISSING_DEPENDENCY  Broken lockfile: no entry for '<first dependency package name>' in pnpm-lock.yaml`

**My current solution:**

Migrated pnpm from native to [corepack](https://nodejs.org/api/corepack.html) and downgraded to `8.5.1`

Hope this ticket will help anyone with this problem.

**Comment by swarnava** [comment]
Hi,

We are investigating this currently and update you with more details.

**Comment by swarnava** [comment]
Hi,

We have pushed a fix for this just now. Could you try redeploying and let us know how it goes?

**Reply by joulev** [reply]
Thanks, it works now 🎉

**Reply by z4nr34l** [reply]
Confirmed, works now.

**Comment by ilbertt** [comment]
I'm having the same issue.

My `package.json`:
```json
{
  ...
  "packageManager": "pnpm@9.0.6",
  ...
}
```

The Vercel env variable:
- `ENABLE_EXPERIMENTAL_COREPACK`: `1`

The error I get in the Vercel build:
```
ERR_PNPM_BAD_PM_VERSION  This project is configured to use v9.0.6 of pnpm. Your current pnpm is v9.0.4
```

I've also tried to change the `packageManager` version to **9.1.1**, but Vercel was still running with **9.0.4** and hence throwing that error.

**Reply by kms0219kms** [reply]
Can you try to set this three envs?

* set the environment variable ENABLE_EXPERIMENTAL_COREPACK to 1 following this guide: [vercel.com/docs/deployments/configure-a-build#corepack](https://vercel.com/docs/deployments/configure-a-build#corepack)
* set env var COREPACK_ENABLE_STRICT to 0
* set package-manager-strict to false in .npmrc

**Comment by kms0219kms** [comment]
Same here. Could you check what's going on now??

**Comment by santanaluiz** [comment]
broken for me too

**Comment by kms0219kms** [comment]
I solved this problem by doing this:
https://github.com/orgs/vercel/discussions/6865#discussioncomment-9468470

**Reply by santanaluiz** [reply]
worked for me too

**Reply by ilbertt** [reply]
This is not a solution, but just a workaround to make it work...

Have a look at this reply too: https://github.com/orgs/vercel/discussions/6865#discussioncomment-9494258

**Comment by swarnava** [comment, marked as answer]
We've disabled the `pnpm@9` strictness generally for Vercel builds. You will still see the pnpm warning in the build logs, but it should no longer fail the build.

If you are still seeing issues with this, please follow up [here](https://github.com/vercel/vercel/issues/11607#issuecomment-2140817961).
